WireDomain/Sources/WireDomain/Repositories/ConversationLabelsRepository.swift

```diff
@@ -0,0 +1,152 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreData
+import WireAPI
+import WireDataModel
+
+/// Facilitate access to conversation labels related domain objects.
+
+protocol ConversationLabelsRepositoryProtocol {
+
+    /// Pull conversation labels from the server and store locally
+
+    func pullConversationLabels() async throws
+}
+
+final class ConversationLabelsRepository: ConversationLabelsRepositoryProtocol {
+
+    private let userPropertiesAPI: any UserPropertiesAPI
+    private let context: NSManagedObjectContext
+
+    init(
+        userPropertiesAPI: any UserPropertiesAPI,
+        context: NSManagedObjectContext
+    ) {
+        self.userPropertiesAPI = userPropertiesAPI
+        self.context = context
+    }
+
+    /// Retrieve from backend and store conversation labels locally
+
+    func pullConversationLabels() async throws {
+        let result = try await userPropertiesAPI.getProperty(forKey: .labels)
```

Comment:
I wonder if it would be nicer to have a method like:

```swift
let remoteLabels = try await userPropertiesAPI.getLabels()
```

That way we don't have to deal with other properties? What do you think?

Reply:
Good idea 👍

Reply:
fixed in last commit, I also provided getter async throwable var for the other properties